legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabTokenManager.java

```diff
@@ -90,12 +94,45 @@ void setGitLabToken(GitLabToken token)
         this.token = token;
     }
 
+    public void setRefreshToken(GitLabToken refreshToken)
+    {
+        if (refreshToken == null)
+        {
+            throw new IllegalArgumentException("token may not be null");
+        }
+        if (refreshToken.getTokenType() == null)
+        {
+            throw new IllegalArgumentException("token type may not be null");
+        }
+        this.refreshToken = refreshToken;
+    }
+
+    public void setTokenExpiry(long expiresIn)
+    {
+        if (expiresIn == 0)
+        {
+            expiresIn = DEFAULT_EXPIRY;
+        }
+        this.tokenExpiry = LocalDateTime.now().plusSeconds(expiresIn -  1800);
+    }
+
+    public GitLabToken getRefreshToken()
+    {
+        return this.refreshToken;
+    }
+
+    public boolean isTokenExpired()
```

Comment:
I think this should be called something like "shouldRefreshToken", since we want to refresh the token before it actually expires.
```suggestion
    public boolean shouldRefreshToken()
```